handler.py

```diff
@@ -271,19 +276,35 @@ def main():
         overlay_files = [x for x in overlay_files]
         overlay_mosaic = create_mosaic(overlay_files, Path(f"{args.output_directory}/mosaics/overlay.tif"))
 
+    # Get files for creating shapefile and/or pushing to AGOL
+    if args.create_shapefile or args.agol_user:
+        dmg_files = get_files(Path(args.output_directory) / 'dmg')
+
     if args.create_shapefile:
         print('Creating shapefile')
-        files = get_files(Path(args.output_directory) / 'dmg')
-        create_shapefile(files,
+        create_shapefile(dmg_files,
                          Path(args.output_directory).joinpath('shapes') / 'damage.shp',
                          args.destination_crs)
 
+    if args.agol_user and args.agol_password and args.agol_feature_service and args.agol_layer_num:
+        agol_polys = create_polys(dmg_files)
+        feat_set = get_feature_set(agol_polys)
+        try:
+            new_feat = agol_append(args.agol_user,
+                                   args.agol_password,
+                                   feat_set,
+                                   args.agol_feature_service,
+                                   args.agol_layer_num)
+            print(f'Appended {new_feat} new features to ArcGIS online.')
+        except Exception as e:
```

Comment:
Don't catch `Exception`. Catch the specific exception. Also, if you're just printing the error anyways, just `raise(e)` instead of printing `e`.

Reply:
My thinking was to catch whatever may be kicked to not crash the program and still get some useful information. Presumably, I would be catching the exception from the API return so I'm not sure the possible exceptions. A better way to do that?